codegen/type_converter.go

```diff
@@ -638,25 +671,78 @@ func (c *TypeConverter) genStructConverter(
 			var (
 				stFromPrefix = keyPrefix
 				stFromType   compile.TypeSpec
+				fromTypeName string
 			)
 			if fromField != nil {
 				stFromType = fromField.Type
 				stFromPrefix = keyPrefix + pascalCase(fromField.Name)
+
+				var err error
+				fromTypeName, err = c.getIdentifierName(stFromType)
+				if err != nil {
+					fromTypeName = ""
+				}
 			}
-			err := c.genConverterForStruct(
-				toField.Name,
-				toFieldType,
-				toField.Required,
-				stFromType,
-				fromIdentifier,
-				keyPrefix+pascalCase(toField.Name),
-				stFromPrefix,
-				indent,
-				fieldMap,
-			)
+
+			toTypeName, err := c.getIdentifierName(toFieldType)
 			if err != nil {
 				return err
 			}
+
+			if converterMethodName, ok := c.convStructMap[toFieldType.Name]; ok {
```

Comment:
this struct case is getting big, can we put it in a helper function for better readability?

Reply:
Yeah good idea!